src/realm/object_id.cpp

```diff
@@ -57,19 +58,22 @@ bool ObjectId::is_valid_str(StringData str) noexcept
 
 ObjectId::ObjectId(const char* init) noexcept
 {
-    char buf[3];
+    char buf[3] = "";
```

Comment:
Now you won't have the terminating '\0' at position 2.

Reply:
From https://en.cppreference.com/w/cpp/language/zero_initialization:

> When an array of any character type is initialized with a string literal that is too short, the remainder of the array is zero-initialized.

Reply:
Well - I have to give you that.